### PR TITLE
2.0.0

### PR DESCRIPTION
**Pull request by Salonso1602**
# References  
- [Release in Jira](https://focusuy.atlassian.net/projects/BMC2/versions/10525/tab/release-report-all-issues)

# Release notes

## Changes

- [BMC2-1478](https://focusuy.atlassian.net/browse/BMC2-1478) Add support for I2C PullUp resistor configuration in SupernovaController

- [BMC2-1507](https://focusuy.atlassian.net/browse/BMC2-1507) Add support to ENTDAA in SupernovaController

- [BMC2-1508](https://focusuy.atlassian.net/browse/BMC2-1508) Handle new reponses for DAA related methods in SupernovaController

- [BMC2-1509](https://focusuy.atlassian.net/browse/BMC2-1509) Update signature for the SETAASA related method in the SupernovaController

- [BMC2-1514](https://focusuy.atlassian.net/browse/BMC2-1514) Improve UART\_ALREADY\_INITIALIZED\_ERROR presentation in SNController

- [BMC2-1626](https://focusuy.atlassian.net/browse/BMC2-1626) Update SNController SDK and FW support to 3.0.0

- [BMC2-1628](https://focusuy.atlassian.net/browse/BMC2-1628) Update SETAASA and SETDASA error handling to present errors to user

- [BMC2-1658](https://focusuy.atlassian.net/browse/BMC2-1658) Update IBI examples to new IBI structure in SDK/FW 3.0.0

- [BMC2-1686](https://focusuy.atlassian.net/browse/BMC2-1686) Address code review comments in basic\_i2c\_example.py

- [BMC2-1691](https://focusuy.atlassian.net/browse/BMC2-1691) Release SNController 2.0.0

- [BMC2-1792](https://focusuy.atlassian.net/browse/BMC2-1792) Extend support for HDR Exit Pattern in Supernova Controller

## Fixes

- [BMC2-1659](https://focusuy.atlassian.net/browse/BMC2-1659) Fix SETXTIME parameter order in SDK call

- [BMC2-1797](https://focusuy.atlassian.net/browse/BMC2-1797) Fix how I3C Init bus checks for errors.

# How to test  
Run all tests  
Run all examples

### CODE REVIEW

**Comment by pons93** [comment]
I3C is not working, apparently we are checking errors with:
`if status == "DAA_SUCCESS" and responses[0]["errors"] == "NO_TRANSFER_ERROR":`
when the response coming is (for an init bus for example):
`[{'id': 8, 'command': 2, 'name': 'I3C INIT BUS', 'result': 'DAA_SUCCESS', 'errors': ['NO_TRANSFER_ERROR'], 'invalidAddresses': []}]`

**Comment by Salonso1602** [comment]
> I3C is not working, apparently we are checking errors with: `if status == "DAA_SUCCESS" and responses[0]["errors"] == "NO_TRANSFER_ERROR":` when the response coming is (for an init bus for example): `[{'id': 8, 'command': 2, 'name': 'I3C INIT BUS', 'result': 'DAA_SUCCESS', 'errors': ['NO_TRANSFER_ERROR'], 'invalidAddresses': []}]`

Fixed @ https://github.com/binhollc/SupernovaController/pull/89

**Review by pons93 — APPROVED** [review]
LGTM!!!

File: tests/i3c_ccc_tests.py [hunk 1]
```diff
@@ -0,0 +1,186 @@
+import os
+import unittest
+from supernovacontroller.sequential.supernova_device import SupernovaDevice
+from supernovacontroller.sequential.i3c import SupernovaI3CBlockingInterface
+
+from binhosimulators import BinhoSupernovaSimulator
+
+BMM350_DATA = {"asString": ["0x07", "0x70", "0x10", "0x33", "0x00", "0x00"], "asInt": [7, 112, 16, 51, 0, 0]}
+BMI323_DATA = {"asString": ["0x07", "0x70", "0x10", "0x43", "0x10", "0x00"], "asInt": [7, 112, 16, 67, 16, 0]}
```
Comment (gusaaaaa) [inline tests/i3c_ccc_tests.py:9]: I hate this and I love it at the same time 🤣

**Comment by gusaaaaa** [comment]
I've got a `HARDWARE_FEATURE_NOT_SUPPORTED` when setting pull-up resistor of 150 Ohm:

```
(.venv) ➜  SupernovaController git:(develop-2.0.0) USE_REAL_DEVICE=True python tests/i2c_tests.py
..F.........
======================================================================
FAIL: test_i2c_set_pull_up_resistor (__main__.TestSupernovaController)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/gusa/Projects/Binho/SupernovaController/tests/i2c_tests.py", line 55, in test_i2c_set_pull_up_resistor
    self.assertTupleEqual((True, 150), (success, message), f"Setting 150 Ohm failed: {message}")
AssertionError: Tuples differ: (True, 150) != (False, ['HARDWARE_FEATURE_NOT_SUPPORTED'])

First differing element 0:
True
False

- (True, 150)
+ (False, ['HARDWARE_FEATURE_NOT_SUPPORTED']) : Setting 150 Ohm failed: ['HARDWARE_FEATURE_NOT_SUPPORTED']

----------------------------------------------------------------------
Ran 12 tests in 5.499s

FAILED (failures=1)
```

**Review by gusaaaaa — APPROVED** [review]
It looks good to me! All tests pass except for one related with I2C pull-ups.

**Comment by Salonso1602** [comment]
> I've got a `HARDWARE_FEATURE_NOT_SUPPORTED` when setting pull-up resistor of 150 Ohm:

That is due to testing with a rev B 
That failure is expected in that case

**Pull request merged**